linker.go

```diff
@@ -457,3 +459,41 @@ func resolveKconfigReferences(insns asm.Instructions) (_ *Map, err error) {
 
 	return kconfig, nil
 }
+
+func resolveKsymReferences(insns asm.Instructions) error {
+	var ksymMap map[string]uint64
+	var err error
+
+	iter := insns.Iterate()
+	var missingKsyms []string
+
+	for iter.Next() {
+		meta, _ := iter.Ins.Metadata.Get(ksymMetaKey{}).(*ksymMeta)
+		if meta == nil {
+			continue
+		}
+		// We load the kallsyms map only when really needed
+		if ksymMap == nil {
+			ksymMap, err = kallsyms.LoadSymbolAddresses()
+			if err != nil {
+				return fmt.Errorf("error whild trying to load kallsyms: %v", err)
+			}
+		}
+
+		addr, found := ksymMap[meta.name]
+		if !found {
+			if !slices.Contains(missingKsyms, meta.name) {
+				missingKsyms = append(missingKsyms, meta.name)
+			}
+		}
```

Comment:
Once we have binding info, please make a case here for weak ksyms. If we can't find them, set its value to `0`.

This allows users to write CO-RE code using `__ksym`s
```
extern void some_symbol __ksym __weak;

if (&some_symbol) {
  // we have a kernel with this symbol, use it.
} else {
  // we are running on a kernel without this symbol, execute fallback code.
}
```